File: src/main/java/com/uapp_llc/repository/ColumnRepository.java
package com.uapp_llc.repository;

import java.util.List;
import java.util.Optional;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import com.uapp_llc.model.Column;

public interface ColumnRepository {

  Column save(Column entity);

  Optional<Column> findById(Long id);

  List<Column> findAll();

  Page<Column> findAll(Pageable pageable);

  int count();

  void delete(Column entity);

}

File: src/main/kotlin/com/uapp_llc/repository/ColumnRepository.kt
package com.uapp_llc.repository

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import com.uapp_llc.model.Column

interface ColumnRepository {

  fun save(entity: Column): Column

  fun findById(id: Long): Column?

  fun findAll(pageable: Pageable): Page<Column>

  fun findAllByOrderByIndex(): List<Column>

  fun count(): Int

  fun delete(entity: Column)

}
